includes/admin/class-order-meta-box.php

```diff
@@ -0,0 +1,263 @@
+
+		if ( empty( $last_sync_timestamp ) ) {
+			if ( empty( $last_error ) ) {
+				$queue_id = TaxJar_Order_Record::find_active_in_queue( $order->get_id() );
+
+				if ( $queue_id ) {
+					return 'Order is currently in the sync queue. ' . self::get_sync_queue_link( $order );
+				}
+
+				$record = new TaxJar_Order_Record( $order->get_id(), true );
+				$record->load_object( $order );
+				$can_sync = $record->should_sync();
+
+				if ( $can_sync ) {
+					return 'Order is ready to sync to TaxJar but has not yet been added to the sync queue.';
+				} else {
+					return 'Order cannot sync to TaxJar. ' . $record->get_error()['message'];
+				}
+			} else {
+				return 'Order failed to sync to TaxJar. ' . $last_error;
+			}
+		} else {
+			$sync_date = wp_date( wc_date_format(), wc_string_to_timestamp( $last_sync_timestamp ) );
+			$sync_time = wp_date( wc_time_format(), wc_string_to_timestamp( $last_sync_timestamp ) );
+			return 'Order successfully synced to TaxJar.<br>Last Synced on: ' . $sync_date . ' ' . $sync_time;
+		}
+	}
+
+	/**
+	 * Get accordion content for a refund.
+	 *
+	 * @param \WC_Order_Refund $refund Refund.
+	 *
+	 * @return string
+	 */
+	public static function get_refund_sync_accordion_content( \WC_Order_Refund $refund ): string {
+		$last_sync_timestamp = $refund->get_meta( '_taxjar_last_sync' );
+		$last_error          = $refund->get_meta( '_taxjar_sync_last_error' );
+
+		if ( empty( $last_sync_timestamp ) ) {
+			if ( empty( $last_error ) ) {
+				$queue_id = TaxJar_Refund_Record::find_active_in_queue( $refund->get_id() );
+
+				if ( $queue_id ) {
+					return 'Refund is currently in the sync queue. ' . self::get_sync_queue_link( $refund );
+				}
+
+				$record = new TaxJar_Refund_Record( $refund->get_id(), true );
+				$record->load_object();
+				$can_sync = $record->should_sync();
+
+				if ( $can_sync ) {
+					return 'Refund is ready to sync to TaxJar but has not yet been added to the sync queue.';
+				} else {
+					return 'Refund cannot sync to TaxJar. ' . $record->get_error()['message'];
+				}
+			} else {
+				return 'Refund failed to sync to TaxJar. ' . $last_error;
+			}
+		} else {
+			$sync_date = wp_date( wc_date_format(), wc_string_to_timestamp( $last_sync_timestamp ) );
+			$sync_time = wp_date( wc_time_format(), wc_string_to_timestamp( $last_sync_timestamp ) );
+			return 'Refund successfully synced to TaxJar.<br>Last Synced on: ' . $sync_date . ' ' . $sync_time;
+		}
+	}
+
+	/**
+	 * Get the link to the sync queue search for a particular order.
+	 *
+	 * @param \WC_Abstract_Order $order Order or Refund to get sync queue link for.
+	 *
+	 * @return string
+	 */
+	private static function get_sync_queue_link( \WC_Abstract_Order $order ): string {
+		return '<a href="' . admin_url( 'admin.php?s=' . $order->get_id() . '&page=wc-settings&tab=taxjar-integration&section=sync_queue&taxjar_record_type&taxjar_record_status&paged=1' ) . '" >View Progress</a>';
```

Comment:
Thoughts?:
![image](https://user-images.githubusercontent.com/47947793/148617030-8e5f4449-e0a4-4cb0-be78-e45cccd9f638.png)

Reply:
I've implemented a form of this improvement. Thanks!